src/main/java/reposense/parser/ArgsParser.java

```diff
@@ -0,0 +1,101 @@
+    private static final String MESSAGE_SINCE_DATE_LATER_THAN_UNTIL_DATE =
+            "\"Since Date\" cannot be later than \"Until Date\"";
+
+    private static ArgumentParser getArgumentParser() {
+        ArgumentParser parser = ArgumentParsers
+                .newFor(PROGRAM_USAGE)
+                .addHelp(false)
+                .build()
+                .description(PROGRAM_DESCRIPTION);
+
+        parser.addArgument("-h", "--help")
+                .help("Show help message.")
+                .action(new HelpArgumentAction());
+
+        parser.addArgument("-config")
+                .required(true)
+                .type(Arguments.fileType().verifyExists().verifyIsFile().verifyCanRead())
+                .metavar("PATH")
+                .help("The path to the CSV config file to read.");
+
+        parser.addArgument("-output")
+                .metavar("PATH")
+                .type(Arguments.fileType().verifyExists().verifyIsDirectory().verifyCanWrite())
+                .setDefault(new File("."))
+                .help("The path to the dashboard generated. "
+                        + "If not provided, it will be generated in the current directory.");
+
+        parser.addArgument("-since")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("The date to start filtering.");
+
+        parser.addArgument("-until")
+                .metavar("dd/MM/yyyy")
+                .type(new DateArgumentType())
+                .setDefault(Optional.empty())
+                .help("The date to stop filtering.");
+
+        return parser;
+    }
+
+    /**
+     * Parses the given string arguments to a {@code CliArguments} object.
+     * @throws ParseException if the given string fails to parse to a CliArguments object.
+     */
+    public static CliArguments parse(String[] args) throws ParseException {
+        try {
+            ArgumentParser parser = getArgumentParser();
+            Namespace results = parser.parseArgs(args);
+
+            File configFile = results.get("config");
+            File outputFile = results.get("output");
+            Optional<Date> sinceDate = results.get("since");
+            Optional<Date> untilDate = results.get("until");
+
+            Path configFilePath = Paths.get(configFile.toURI());
+            Path outputFilePath = Paths.get(outputFile.toURI());
+
+            verifyDatesRangeIsCorrect(sinceDate, untilDate);
+            return new CliArguments(configFilePath, outputFilePath, sinceDate, untilDate);
+        } catch (ArgumentParserException ape) {
+            throw new ParseException(String.format(getArgumentParser().formatUsage() + ape.getMessage()));
+        }
+    }
+
+    /**
+     * Verifies that {@code sinceDate} is earlier than {@code untilDate}.
+     * @throws IllegalArgumentException if user-supplied values are invalid.
```

Comment:
`@throws IllegalArgumentException if {@code sinceDate} supplied is later than {@code untilDate}`
if the supplied values are invalid, we would get a `ParseException` doing `parse` instead right?